plasmapy/plasma/grids.py

```diff
@@ -1118,12 +1118,23 @@ def volume_averaged_interpolator(
             return tuple(output)
 
 
-class NonUniformCartesianGrid(CartesianGrid):
+class NonUniformCartesianGrid(AbstractGrid):
     r"""
     A Cartesian grid in which the _make_mesh method produces a non-uniformly
     spaced grid.
     """
 
+    def _validate(self):
+        # Check that all units are lengths
+        for i in range(3):
+            try:
+                self.units[i].to(u.m)
+            except u.UnitConversionError:
+                raise ValueError(
```

Comment:
Could do with a test for this one! If it's not too much effort, that is.

Reply:
Oops, not at all (already had it implemented for CartesianGrid)